qtfred/src/mission/dialogs/CampaignEditorDialogModel.cpp

```diff
@@ -0,0 +1,801 @@
+#include "CampaignEditorDialogModel.h"
+#include "weapon/weapon.h"
+#include "cutscene/cutscenes.h"
+#include "menuui/mainhallmenu.h"
+#include "stats/scoring.h"
+#include "mission/missiongoals.h"
+#include "mission/missionsave.h"
+
+#include <QPlainTextDocumentLayout>
+
+extern int Skip_packfile_search;
+
+extern void parse_init(bool basic);
```

Comment:
Where are these functions defined?  If they're elsewhere in a .cpp *without* a .h declaring them, then this must be considered very carefully in its design as well as have a comment pointing to where they are actually defined.  There's also the potential issue of namespace collisison with the usage of `extern` in this manner.

Reply:
`Skip_packfile_search` is from cfilesystem.cpp, I'm using it to find only loose/editable missions, like oldfred did.
The rest are from missionparse.cpp. `parse_init` is also used by the qtfred initialisation when opening a mission, like the rest I'm using it in parseMnPart, to parse relevant parts of available missions _only_. Added comments to clarify.